docs/web/sdk-language-strings.mdx

```diff
@@ -0,0 +1,59 @@
+---
+title: Language Strings
+description: Geocortex Web - Learn about internationalization and language strings in Geocortex Web
+---
+
+import PrereqsCustomCodeSnippet from "./snippets/prereqs-custom-code.mdx";
+import TranslationCallout from "./snippets/translation-callout.mdx";
+
+<TranslationCallout />
+
+Geocortex Web was designed to be fully translatable on all levels of the product, from app config to custom code.
+
+Language strings provide a couple advantages over hardcoding plain text in a component.
+
+1. Language strings can be easily translated or modified without having to change component code.
+2. Language strings can be shared across components.
+3. Language strings consolidate user facing text throughout the application, making it easy to review and find.
+
+It is recommended you use language strings for all user facing text in your application.
+
+## Defining Language Strings
+
+Language strings are defined as properties in a plain JSON file. It's convention to define language strings with the format `language-kebab-cased-key`.
+
+```json title="en.json"
+{
+    "language-some-text-string": "I will be translated."
+}
+```
+
+:::warning
+If you define a language string that Geocortex Web has already defined, it will override the existing language string with your custom string.
+:::
+
+By convention, these files are named after the locale they represent. For example, a file of german language strings would be named `de.json`.
+
+## Registering Language Resources
+
+All language resource files need to be registered with the SDK before they can be used.
+
+:::note
+Geocortex Web uses two letter language codes to identify locales. See the [ISO 639-1 codes](https://en.wikipedia.org/wiki/List_of_ISO_639-1_codes) for a full list of possible locales.
```

Comment:
Not sure that this is technically accurate as i believe we support the country code variants of the language codes: `en-US` vs `en`.

This can be a fair bit more common in Europe where you have different variants of languages such as German/Swiss German.

@eric-parsons i know we had shifted a bit in how we present these language codes in designer. How should we talk about this here do you thiink?

Reply:
At the SDK level, I think we should talk about what Web actually supports, which is full support for any language tags, as per https://www.w3.org/International/questions/qa-choosing-language-tags. Note that in the above link it talks about casing being a convention, but for us it's actually required. All lowercase is a language code, all uppercase is a region code, and Pascal case is a script. The restriction of showing only language codes is a Designer UI thing and has nothing to do with Web.

However, it's might be important to document how our resolution works. If you configure language data in the "de-AT" locale exclusively and someone comes along with a browser configured as "de", they will see the invariant translation (or "fallback" locale if that's configured), not "de-AT". The potential for confusion is why Designer was changed to only show language codes in the locale dropdown.

Reply:
Added a section on "Supported Locales" below this section with the information provided by @eric-parsons